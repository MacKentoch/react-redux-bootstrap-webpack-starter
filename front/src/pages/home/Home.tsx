import React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';
import HomeInfo from './styled/HomeInfo';
import MainTitle from './styled/MainTitle';
import { ReduxConnectedProps, OwnProps } from './index';

// #region  types
export type Props = RouteComponentProps & ReduxConnectedProps & OwnProps;
// #endregion

function Home() {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <HomeInfo>
          <MainTitle>ReactJS 16.14 Bootstrap 4</MainTitle>
          <h2>
            with Hot Reload (<i>react-hot-loader 4+</i>
            )!!!
          </h2>
          <h2>and React Router v5</h2>
          <h2>and webpack 5.x</h2>
          <h1>Starter</h1>
          <p>
            <Link className="btn btn-success btn-lg" to={'/about'}>
              <i className="fa fa-info" />
              &nbsp; go to about
            </Link>
          </p>
        </HomeInfo>
      </Jumbotron>
    </FadeInEntrance>
  );
}

Home.displayName = 'Home';

export default Home;
