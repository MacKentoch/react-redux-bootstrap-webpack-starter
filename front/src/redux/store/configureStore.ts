import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createBrowserHistory as createHistory } from 'history';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../modules/reducers';

// #region constants
const isProd = process.env.NODE_ENV === 'production';
export const history = createHistory();
// #endregion

// #region createStore : enhancer

// #region logger middleware (dev only)
const loggerMiddleware = createLogger({
  level: 'info',
  collapsed: true,
});
// #endregion

const enhancer = !isProd
  ? composeWithDevTools(
      applyMiddleware(
        thunkMiddleware,
        loggerMiddleware, // logger at the end
      ),
    )
  : composeWithDevTools(applyMiddleware(thunkMiddleware));
// #endregion

// #region persisted reducer
const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['router'],
  // whitelist: ['userAuth'],
};

const persistedReducer = persistReducer(persistConfig, reducer);
// #endregion

export default function configureStore(initialState = {}) {
  const store = createStore(persistedReducer, initialState, enhancer);
  const persistor = persistStore(store);

  if ((module as any).hot) {
    (module as any).hot?.accept('../modules/reducers', () => {
      // @ts-ignore
      const nextAppReducer = require('../modules/reducers').default;
      store.replaceReducer(persistReducer(persistConfig, nextAppReducer));
    });
  }

  return { store, persistor };
}
