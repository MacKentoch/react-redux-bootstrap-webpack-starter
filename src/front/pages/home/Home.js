// @flow

// #region imports
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import { Link } from 'react-router-dom';
import Jumbotron from '../../components/jumbotron/Jumbotron';
import HomeInfo from './styled/HomeInfo';
import MainTitle from './styled/MainTitle';
import LightNote from './styled/LightNote';
// #endregion

// #region flow types
export type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};

export type State = any;
// #endregion

class Home extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <Jumbotron>
          <HomeInfo>
            <MainTitle>ReactJS 16.3+ Redux Bootstrap</MainTitle>
            <h2>
              with Hot Reload (<i>react-hot-loader 4+</i>)!!!
            </h2>
            <h2>and React Router v4</h2>
            <h2>and webpack 4.x</h2>
            <h2>
              and styled-components (<LightNote>
                so keep using SCSS like styles and benefit performant css-in-js
              </LightNote>)
            </h2>
            <h1>Starter</h1>
            <p>
              <Link className="btn btn-success btn-lg" to={'/about'}>
                <i className="fa fa-info" />
                &nbsp; go to about
              </Link>
            </p>
          </HomeInfo>
        </Jumbotron>
      </div>
    );
  }
  // #endregion
}

export default Home;
