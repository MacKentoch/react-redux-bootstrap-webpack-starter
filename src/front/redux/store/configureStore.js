// @flow

/* eslint no-process-env:0 */
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./configureStore.prod');
} else {
  module.exports = require('./configureStore.dev');
}
