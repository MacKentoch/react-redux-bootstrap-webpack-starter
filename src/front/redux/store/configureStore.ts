/* eslint no-process-env:0 */
let configureStore = null;

if (process.env.NODE_ENV === 'production') {
  configureStore = require('./configureStore.prod').default;
  // const {history} = require('./configureStore.prod');
  // export history;
} else {
  configureStore = require('./configureStore.dev').default;
  // const {history} = require('./configureStore.dev');
  // export history;
}
export default configureStore;
