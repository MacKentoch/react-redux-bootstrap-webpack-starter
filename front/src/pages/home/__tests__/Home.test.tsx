import React from 'react';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import Home from '../index';

const middlewares: Array<any> = [];
const mockStore = configureStore(middlewares);

describe('Home page', () => {
  let rootElement: any = null;

  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <ThemeProvider theme={{}}>
          <MemoryRouter>
            <Home />
          </MemoryRouter>
        </ThemeProvider>
      </Provider>,
      rootElement,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
