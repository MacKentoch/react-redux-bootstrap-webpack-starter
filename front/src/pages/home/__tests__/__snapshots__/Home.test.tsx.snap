// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders as expected 1`] = `
<div
  class="sc-bdVaJa jxVqOJ"
>
  <div
    class="jumbotron"
  >
    <div
      class="sc-bwzfXH uTxCW"
    >
      <h1
        class="sc-htpNat hljmAh"
      >
        ReactJS 16.14 Bootstrap 4
      </h1>
      <h2>
        with Hot Reload (
        <i>
          react-hot-loader 4+
        </i>
        )!!!
      </h2>
      <h2>
        and React Router v5
      </h2>
      <h2>
        and webpack 5.x
      </h2>
      <h1>
        Starter
      </h1>
      <p>
        <a
          class="btn btn-success btn-lg"
          href="/about"
        >
          <i
            class="fa fa-info"
          />
          Â  go to about
        </a>
      </p>
    </div>
  </div>
</div>
`;
